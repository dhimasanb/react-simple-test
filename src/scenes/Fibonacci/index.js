import React, { Component } from "react";
import * as math from "../../utils/math";
import FormComponent from "../../components/Form";

class Fibonacci extends Component {
  constructor(props) {
    super(props);

    this.state = {
      amount: "",
      result: [],
      error: ""
    };
  }

  // Event update the react state when user types
  handleChange = name => event => {
    this.setState({ [name]: event.target.value });
  };

  calculate = () => {
    const value = this.state.amount;
    const result = math.fibonacciSequence(value);

    // Remove last element
    result.pop();

    const joinComma = result.join();

    return this.setState({
      result: joinComma
    });
  };

  render() {
    const { amount, result } = this.state;

    return (
      <FormComponent
        title="Calculator Fibbonacci"
        placeholder="Please input fibbonaci number. Ex: 4 = 0, 1, 1, 2"
        value={amount}
        onClick={() => this.calculate()}
        result={result}
        handleChange={this.handleChange("amount")}
        handleKeyPressEnter={this.calculate}
      />
    );
  }
}

export default Fibonacci;
