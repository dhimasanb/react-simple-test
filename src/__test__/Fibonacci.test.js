import React from "react";
import { shallow } from "enzyme";
import Fibonacci from "../scenes/Fibonacci";
import FormComponent from "../components/Form";

describe("Fibonacci Scene", () => {
  const wrapper = shallow(<Fibonacci />);
  const formWrapper = wrapper.find(FormComponent).shallow();

  // Snapshot test
  it("Renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  // Initializes state test
  it("Initializes data in `state`", () => {
    expect(wrapper.state()).toEqual({
      amount: "",
      result: [],
      error: ""
    });
  });

  // Input and change state
  describe("When typing into the amount input 4", () => {
    const amount = "4";

    beforeEach(() => {
      formWrapper.find("#input-amount").simulate("change", {
        target: { value: amount }
      });
    });

    it("Updates the amount in `state`", () => {
      expect(wrapper.state().amount).toEqual(amount);
    });
  });

  // Click button test
  describe("Click button calculate", () => {
    const result = "0,1,1,2";

    beforeEach(() => formWrapper.find("#btn-calculate").simulate("click"));

    it(`Result is ${result}`, () => {
      expect(wrapper.state().result).toEqual(result);
    });
  });

  // Enter keypress test
  describe("Press enter key", () => {
    const result = "0,1,1,2";

    beforeEach(() =>
      formWrapper.find("#input-amount").simulate("keypress", { key: "Enter" })
    );

    it(`Result is ${result}`, () => {
      expect(wrapper.state().result).toEqual(result);
    });
  });
});
