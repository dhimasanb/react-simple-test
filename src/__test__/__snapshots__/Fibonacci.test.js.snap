// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fibonacci Scene Renders properly 1`] = `ShallowWrapper {}`;
